### DISCUSSION TITLE
Does this plugin work in a cluster setup?

### DISCUSSION THREAD

**Discussion by jlaute** [FriendsOfShopware/FroshPlatformThumbnailProcessor#122] [Q&A] [answered]
First of all, thank you for that great plugin.

I want to know if this plugin works in an environment with multiple app servers, because the `GeneratorCompilerPass` generates the two classes `FileSaver` and `ThumbnailService` on runtime. 

Are these classes also created for each app server when there is a redis cache activated?

**Comment by tinect** [comment, marked as answer]
I don‘t see any reason why it should not work on cluster environments.
These files are generated while the core is being booted/started/… by running e.g. „bin/console“ 

Additionally, PHP-Files are not stored in Redis, are they?

**Reply by jlaute** [reply]
You are probably right. I was just concerned if this boot process is called on each app server on it's own when there is already a cache created on a different server. But without the boot process, the whole DependencyInjection will not work, so it should be quite obvious that this process is triggered automatically on each server, whether there is a cache or not. 

Thx for answering. I will give it a try :)
